components/overledger/overledger.app.mjs

```diff
@@ -11,35 +29,55 @@ export default {
     },
   },
   methods: {
-    _baseUrl() {
-      return "https://api.overledger.io";
-    },
     _headers() {
       return {
         "Authorization": `Bearer ${this.$auth.oauth_access_token}`,
         "Content-Type": "application/json",
         "API-Version": "3.0.0",
       };
     },
+    _getBaseUrl() { //conditional for environment url selection.
+      return this.environment === "sandbox"
+        ? "https://api.sandbox.overledger.io"
+        : "https://api.overledger.io";
+    },
     _makeRequest({
-      $ = this, path, ...otherOpts
+      $ = this, baseUrl, path, ...otherOpts
     }) {
       return axios($, {
         ...otherOpts,
-        url: this._baseUrl() + path,
+        url: baseUrl + path,
         headers: this._headers(),
       });
     },
     prepareSmartContractTransaction(opts = {}) {
       return this._makeRequest({
         method: "POST",
+        baseUrl: this._getBaseUrl(),
```

Comment:
This repetition is unnecessary, the method already receives the environment in the opts param